feat: Support additional filters

kamal opened the pull request:
## Describe Your Changes

Add a filter modal in Hub to house additional filters.

<img width="2652" height="1506" alt="Screenshot 2025-09-09 at 4 01 07 PM" src="https://github.com/user-attachments/assets/db95f7bd-0aca-4f31-b222-201fb52f5f06" />



## Fixes Issues

- Closes #
- Closes #

## Self Checklist

- [ ] Added relevant comments, esp in complex areas
- [ ] Updated docs (for bug fixes / features)
- [ ] Created issues for follow-up changes or refactoring needed

<!-- ELLIPSIS_HIDDEN -->

----

> [!IMPORTANT]
> Add `ModelFilters` component to support filtering models by download status and tool calling capability, with tests and localization updates.
> 
>   - **Components**:
>     - Add `ModelFilters` component in `ModelFilters.tsx` to manage filters for `showOnlyDownloaded` and `toolCallingOnly`.
>     - Integrate `ModelFilters` in `index.tsx` to apply filters to model list.
>   - **Tests**:
>     - Add tests in `ModelFilters.test.tsx` to verify filter toggling and clearing functionality.
>   - **Localization**:
>     - Update `hub.json` in multiple locales to include new filter-related strings: `filters`, `filterBy`, `toolCalling`, `clearFilters`.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=menloresearch%2Fjan&utm_source=github&utm_medium=referral)<sup> for 2ee4fa22aa4384f3d09e3e963c48f1e50ffac37a. You can [customize](https://app.ellipsis.dev/menloresearch/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

<!-- ELLIPSIS_HIDDEN -->

louis-menlo commented on web-app/src/routes/hub/index.tsx:
```diff
@@ -96,6 +101,17 @@ function Hub() {
   const llamaProvider = getProviderByName('llamacpp')
 
+  // Helper function to check if a model supports tool calling
+  const supportsToolCalling = useCallback((model: CatalogModel) => {
+    // Check if the model name matches any of the known tool-supported models
+    return Object.values(DefaultToolUseSupportedModels).some((supportedModel) =>
+      model.model_name.toLowerCase().includes(supportedModel.toLowerCase()) ||
+      model.quants.some((quant) =>
+        quant.model_id.toLowerCase().includes(supportedModel.toLowerCase())
+      )
+    )
+  }, [])
```
We added tool support detection on dev branch, also removed this hard coded DefaultToolUseSupportedModels enum. Could you please help rebase and apply new filter? Thanks.

kamal commented:
Is there anything else you'd like to see?

kamal commented:
@urmauur I've added vision to the list of filters and updated the screenshot.

> searching the Hugging Face repo only works if the downloaded filter is set to false

Can you explain this a bit more? I don't see any change of behavior against the `dev` branch. I can type in the search bar even if the downloaded filter is true/false.

kamal commented:
@urmauur thanks for the explanation. someone on the team will have to merge it.

urmauur commented:
> e explanation. someone on the team will have to merge

help me, fix conflict then will merge this. sorry

urmauur commented:
Thanks a lot for your contribution and the effort you’ve put into this PR @kamal  🙏
After reviewing our current roadmap, we’ve decided to revamp the Hub scree, including its overall UI, layout, UX, and flow. Because of this broader redesign, the changes in this PR will be replaced as part of that upcoming update.

We really appreciate your time and initiative here.  Your contributions are always welcome, and we’d love to have your input once the new Hub structure is ready. Thanks again for understanding 💪
